pkg/sql/colexec/external/external.go

```diff
@@ -360,6 +360,7 @@ func makeBatch(param *ExternalParam, plh *ParseLineHandler, mp *mpool.MPool) *ba
 		typ := types.New(types.T(param.Cols[i].Typ.Id), param.Cols[i].Typ.Width, param.Cols[i].Typ.Scale, param.Cols[i].Typ.Precision)
 		vec := vector.NewOriginal(typ)
 		vector.PreAlloc(vec, batchSize, batchSize, mp)
+		vec.Nsp = nulls.NewWithSize(plh.batchSize)
```

Comment:
It shouldn't be handled that way. If you don't need so much memory, why allocate so much at once?